packages/block-editor/src/store/private-selectors.js

```diff
@@ -45,9 +45,8 @@ export function getLastInsertedBlocksClientIds( state ) {
 export const isBlockSubtreeDisabled = createSelector(
 	( state, clientId ) => {
 		const isChildSubtreeDisabled = ( childClientId ) => {
-			const mode = state.blockEditingModes.get( childClientId );
 			return (
-				( mode === undefined || mode === 'disabled' ) &&
+				getBlockEditingMode( state, childClientId ) === 'disabled' &&
```

Comment:
Just checking, is `undefined` check not required anymore? 

`getBlockEditingMode()` can still try to fetch from the `blockEditingModes` Map:  `state.blockEditingModes.get( childClientId )` which will return `undefined`.

Just by reading the previous check, it seems to say, "if the parent's mode is not yet  known, or has never been set, flag the tree as disabled if all the children are disabled"

Then again, I went through the test description of the [PR that introduced the selector](https://github.com/WordPress/gutenberg/pull/51079), and things still work as expected.

What do you think?

Reply:
I read through all PRs that touched this code and couldn't find the exact reason why the first check used the `getBlockEditingMode` selector and the child subtree did not.

> Just by reading the previous check, it seems to say, "if the parent's mode is not yet known, or has never been set, flag the tree as disabled if all the children are disabled"

This was causing false positives in the case. The paragraph mode is `undefined`, but the check for its children will also resolve to `true` because `[].every( condition ) === true`.

Reply:
> This was causing false positives in the case. The paragraph mode is undefined, but the check for its children will also resolve to true because [].every( condition ) === true.

Ah okay. I just wasn't sure whether a block tree whose children are disabled but whose parent isn't could be classed as "disabled". The function name implies it, but as you say there'd have to be further checks. If it's not being used this way, then it seems safe.

Thanks for confirming!